op-chain-ops/genesis/config.go

```diff
@@ -1044,15 +1048,17 @@ func (d *DeployConfig) RollupConfig(l1StartBlock *eth.BlockRef, l2GenesisBlockHa
 		return nil, errors.New("SystemConfigProxy cannot be address(0)")
 	}
 
-	var soulGasTokenBlock *uint64
+	l1StartTime := l1StartBlock.Time
+
+	var soulGasTokenTime *uint64
 	if d.UseSoulGasToken {
```

Comment:
If `SoulGasTokenTimeOffset` is a nil-able pointer, can we remove `d.UseSoulGasToken`?

Reply:
It'll fail like this: `panic: failed to apply pipeline: error in pipeline stage apply: failed to call L2Genesis script: failed to run L2 genesis script: execution reverted at 0xfdAf657f7146c1AB9db4d70cEd5FDEa7dC4906c0 with message: unrecognized 4 byte signature: 23fd937d`.

I think it's because it's calling into golang precompile [here](https://github.com/QuarkChain/optimism/blob/op-es/op-deployer/pkg/deployer/opcm/l2genesis.go#L50). The latest code doesn't have this precompile([src](https://github.com/ethereum-optimism/optimism/blob/develop/op-deployer/pkg/deployer/opcm/l2genesis.go)), maybe we can try to eliminate it after merge.